### PR TITLE
Added Conditional Semantics to group member query to always include the user.

### PR DESCRIPTION
**Pull request by TheHollidayInn**
Added Conditional Semantics to group member query to always include the user.

This should force the members to always include the user thus fixing #4798 .

You can read about Conditional Semantics here: https://www.safaribooksonline.com/library/view/mongodb-the-definitive/9781449344795/ch04.html .

### CODE REVIEW

File: website/src/controllers/groups.js [hunk 1]
```diff
@@ -30,13 +30,15 @@ var guildPopulate = {path: 'members', select: nameFields, options: {limit: 15} }
...
+ * @param additionalFields: if we want to populate some additional field not fetched normally
  *        pass it as a string, parties only
  */
-var populateQuery = function(type, q, additionalFields){
+var populateQuery = function(type, q, additionalFields, user){
   if (type == 'party')
     q.populate('members', partyFields + (additionalFields ? (' ' + additionalFields) : ''));
   else
+    //Use Conditional Semantics to always include the user
+    guildPopulate.match =  {"_id": { "$ne" : null, "$in": [user._id] } };
```
Comment (crookedneighbor) [inline website/src/controllers/groups.js]: Can you confirm that public guilds you visit when you are not a member of them still show a members list?

**Comment by crookedneighbor** [comment]
@TheHollidayInn You've been rocking the PRs! :+1:

**Comment by TheHollidayInn** [comment]
@crookedneighbor Thanks! I have been having fun.

So, I'm not sure how to visit a guild you aren't a member of, but here is the Public guild page with a user that is a member and one that is not.
screen shot 2015-03-24 at 8 52 57 am
screen shot 2015-03-24 at 8 53 52 am

Also, I added a new commit to prevent errors.

**Comment by crookedneighbor** [comment]
The only way to get to a public guild without being a member is to follow a direct link (IE, someone in the tavern says, "Join the newbies guild!" and links to the guild). To test that, join a guild, copy the URL for the guild in your browser's address bar, leave the guild and then go to the url.

**Comment by TheHollidayInn** [comment]
screen shot 2015-03-24 at 9 00 34 am

Confirmed!

**Comment by Alys** [comment]
Tested. Seems good! 

@TheHollidayInn Thank you! This has been recorded in your account as partial credit towards tier 4.

**Comment by Alys** [comment]
Sorry @TheHollidayInn I have reverted this temporarily because I am seeing odd behaviour in the header when the test player is in a party. The avatars do not load fully. Hopefully we can sort this out and merge again!

**Comment by TheHollidayInn** [comment]
Ah, I believe this is because I set additional fields to null by default. I'll investigate, and then I will verify all the calls made to the modified function. Sorry for this :(

**Comment by Alys** [comment]
It's cool! :)  This is a complex site with a lot of things interrelating. It's common to have glitches where a new feature affects something that seems unrelated. I'm still pretty terrible at working out which other features to look at when I'm testing new features. Noticing this bug was nothing but luck. :-/

**Comment by crookedneighbor** [comment]
once I have a local install again, I'm hoping to write tests around these kinds of things to catch bugs like this.

**Pull request merged**